### DISCUSSION TITLE
More complex secrets instead of only regex [a-zA-Z0-9-]+

### DISCUSSION THREAD

**Discussion by tobiloeb** [NiklasMerz/cordova-plugin-fingerprint-aio#287] [General]
### Feature Request
Is it possible to allow more complex secrets? The very simple allowed secrets of regex [a-zA-Z0-9-]+ are to weak for a good password to encrypt and store.

#### Feature Description
A complex password is state of the art. So the secret property should be able to consume complex strings with including special characters (UTF-8).

Btw great update for this plugin! 👍 Thanks

**Comment by NiklasMerz** [comment]
You could try to encode the password before saving. The secret storage is more intended to store tokens (like OAuth) instead of passwords though.

A longer password is better than a complex short password AFAIK ;-)

**Comment by Dahmsk** [comment]
I use B64 to encrypt my passwords and onyl store the encrypted string. But I would also welcome the feature to store more complex strings as Base64 is not unicode complete.

**Comment by leolio86400** [comment]
Hi guys, just letting you know that the `[a-zA-Z0-9-]+` pattern is more an indication than something else.

It happens that when trying to store some information in the secret some people had some issues with more complex strings on Android.

Feel free to investigate how to support a wider range of characters, the code is in `src/android/CryptographyManagerImpl.java`

```
@Override
    public EncryptedData encryptData(String plaintext, Cipher cipher) throws CryptoException {
        try {
            byte[] ciphertext = cipher.doFinal(plaintext.getBytes(StandardCharsets.UTF_8));
            return new EncryptedData(ciphertext, cipher.getIV());
        } catch (Exception e) {
            throw new CryptoException(e.getMessage(), e);
        }
    }

    @Override
    public String decryptData(byte[] ciphertext, Cipher cipher) throws CryptoException {
        try {
            byte[] plaintext = cipher.doFinal(ciphertext);
            return new String(plaintext, StandardCharsets.UTF_8);
        } catch (Exception e) {
            throw new CryptoException(e.getMessage(), e);
        }
    }
```
